### PR TITLE
Konnector for captain train 

### LINKED ISSUE #178: Konnector for Captain Train

### PR DESCRIPTION
**Pull request by ZeHiro**
Fixes #178 
Buying tickets on Captain Train can lead to different operation labels: 'Capitain Train', 'Capitaine train' (for older operations), for Ouigo tickets bought on capitaine train, the operation is directly labelled 'Ouigo'. I then allowed to use an array as the operation identifier to allow to manage these situations. To maintain compatity with existing konnectors,  if the passed label is a string, it is put inside an array.

Capitain Train connector can also import refund bills, to be sure the bills are matched to the appropriate operation, in case on the same day, a train ticket is bought, and the same amount is refounded, the distinction was made between normal payments, and refunds. Refunds need do be declared as such, using isRefund attribute of the Bill object. 

Still known problem : if exacty the same operation is done on the same day (by twice the same ticket with same amount ), one of the two operations won't be linked to a bill, but the bill will be imported.

### CODE REVIEW

**Comment by Chocobozzz** [comment]
IMHO, you should `return` when calling the callback on error. I think it is worth to activate [this lint rule](http://eslint.org/docs/rules/callback-return) in ESLint.

**Comment by ZeHiro** [comment]
Makes sense.

File: server/lib/link_bank_operation.coffee [hunk 1]
```diff
@@ -49,21 +52,32 @@ class BankOperationLinker
 
         try
             amount = Math.abs parseFloat entry.amount
+            # By default, an entry is an expense. If it is not, it should be
+            # declared as a refund: isRefund=true.
+            if not entry.isRefund? || not entry.isRefund
```
Comment (frankrousseau) [inline server/lib/link_bank_operation.coffee]: You could write:

```coffeescript
amount *= -1 if  entry.isRefund? and entry.isRefund
```

**Comment by frankrousseau** [comment]
Could you explain what you are doing with the pattern matching?

**Comment by ZeHiro** [comment]
I'm not sure i understand the question :s

**Comment by frankrousseau** [comment]
yes and tell us what the changes do.

**Comment by ZeHiro** [comment]
I have updated the comment of the PR.

File: server/lib/link_bank_operation.coffee (hunk 1 above)
Comment (ZeHiro) [inline server/lib/link_bank_operation.coffee]: Yep

**Comment by frankrousseau** [comment]
Thx @ZeHiro

**Pull request merged**